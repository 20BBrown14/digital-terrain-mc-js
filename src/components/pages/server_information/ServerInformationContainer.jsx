import React from 'react';
import ServerInformationView from './ServerInformationView';

/**
 * Server information page container.
 */
const ServerInformationContainer = () => (
  <ServerInformationView />
);

export default ServerInformationContainer;
