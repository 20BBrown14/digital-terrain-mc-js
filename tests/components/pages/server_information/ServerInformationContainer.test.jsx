import React from 'react';
import ServerInformationContainer from '../../../../src/components/pages/server_information/ServerInformationContainer';
import serverInformationService from '../../../../src/components/pages/server_information/service';

jest.mock('../../../../src/components/pages/server_information/service')

describe('ServerInformationContainer', () => {
  it('renders a default view', () => {
    const container = render(<ServerInformationContainer />);
    expect(container).toMatchSnapshot();
  });

  it('has a ServerInformationView', () => {
    const container = shallow(<ServerInformationContainer />);
    expect(container.find('ServerInformationView')).toHaveLength(1);
  });

  it('has a default state', () => {
    const container = shallow(<ServerInformationContainer />).instance();
    const expectedState = {
      isLoading: true,
      hasServiceFailure: false,
      serverInformation: {},
    };

    expect(container.state).toEqual(expectedState);
  });

  describe('componentDidMount', () => {
    it('calls server information service', () => {
      const container = shallow(<ServerInformationContainer />).instance();
      serverInformationService.mockClear();

      container.componentDidMount();
      expect(serverInformationService).toHaveBeenCalledTimes(1);
    });

    describe('successful service call', () => {
      it('sets state as expected', () => {
        const container = shallow(<ServerInformationContainer />).instance();
        serverInformationService.mockClear();
        serverInformationService.mockImplementation((success) => {
          success({ serverInformation: [] });
        });
        container.state.isLoading = true;
        container.componentDidMount();

        expect(container.state.isLoading).toEqual(false);
        expect(container.state.serverInformation).toEqual({ serverInformation: [] });
      });
    });

    describe('failed service call', () => {
      it('sets state as expected', () => {
        const container = shallow(<ServerInformationContainer />).instance();
        serverInformationService.mockClear();
        serverInformationService.mockImplementation((success, failure) => {
          failure();
        });
        container.state.isLoading = true;
        container.componentDidMount();

        expect(container.state.isLoading).toEqual(false);
        expect(container.state.hasServiceFailure).toEqual(true);
      });
    });
  });
});
