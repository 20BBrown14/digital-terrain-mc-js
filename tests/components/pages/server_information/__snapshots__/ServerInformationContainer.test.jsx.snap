// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerInformationContainer renders a default view 1`] = `
<div>
  Server Information Page
</div>
`;
